### PR TITLE
Eval harness

### PR DESCRIPTION
**Pull request by Giuseppe5**
## Risk Highlight

- [ ] This PR includes code from another work (please detail).
- [ ] This PR contains API-breaking changes.
- [ ] This PR depends on work in another PR (please provide links/details).
- [ ] This PR introduces new dependencies (please detail).
- [ ] There are coverage gaps not covered by tests.
- [ ] Documentation updates required in subsequent PR.

## Checklist

- [ ] Code comments added to any hard-to-understand areas, if applicable.
- [ ] Changes generate no new warnings.
- [ ] Updated any relevant tests, if applicable.
- [ ] No conflicts with destination `dev` branch.
- [ ] I reviewed my own code changes.
- [ ] Initial CI/CD passing.
- [ ] 1+ reviews given, and any review issues addressed and approved.
- [ ] Post-review full CI/CD passing.

### CODE REVIEW

File: src/brevitas_examples/llm/main.py [hunk 1]
```diff
@@ -462,6 +481,28 @@ def main(args):
...
+            wrapped_model = HFLM(pretrained=model)  # need to wrap for LLM eval
+            results = evaluator.simple_evaluate(
+                model=wrapped_model,
+                model_args=None,
+                tasks=list(args.zero_shot_tasks),
+                device='cuda:0',
+                limit=None,
+                num_fewshot=0,
+                log_samples=False,
+                batch_size="auto:3",
```
Comment (nickfraser) [inline src/brevitas_examples/llm/main.py:502]: `batch_size=None`?

File: src/brevitas_examples/llm/main.py [hunk 2]
```diff
@@ -462,6 +481,28 @@ def main(args):
...
+                model = torch.compile(model)
+
+            wrapped_model = HFLM(pretrained=model)  # need to wrap for LLM eval
+            results = evaluator.simple_evaluate(
+                model=wrapped_model,
+                model_args=None,
+                tasks=list(args.zero_shot_tasks),
+                device='cuda:0',
+                limit=None,
+                num_fewshot=0,
```
Comment (nickfraser) [inline src/brevitas_examples/llm/main.py:500]: Flag for `None` or `0`?

**Review by nickfraser — APPROVED** [review]
LGTM, but with comments ;)

File: src/brevitas_examples/llm/main.py [hunk 3]
```diff
@@ -462,6 +481,28 @@ def main(args):
...
+                model.cuda()
+                model = torch.compile(model)
+
+            wrapped_model = HFLM(pretrained=model)  # need to wrap for LLM eval
+            results = evaluator.simple_evaluate(
+                model=wrapped_model,
+                model_args=None,
+                tasks=list(args.zero_shot_tasks),
+                device='cuda:0',
+                limit=None,
```
Comment (nickfraser) [inline src/brevitas_examples/llm/main.py:499]: Add flag `None` or `int`.

**Comment by nickfraser** [comment]
Regenerate `README` and `config/default/yml`.

File: src/brevitas_examples/llm/main.py [hunk 4]
```diff
@@ -462,6 +481,28 @@ def main(args):
...
+                remove_hooks(model)
+                model.cuda()
+                model = torch.compile(model)
+
+            wrapped_model = HFLM(pretrained=model)  # need to wrap for LLM eval
+            results = evaluator.simple_evaluate(
+                model=wrapped_model,
+                model_args=None,
+                tasks=list(args.zero_shot_tasks),
+                device='cuda:0',
```
Comment (nickfraser) [inline src/brevitas_examples/llm/main.py:498]: Should this be `auto` for accelerate?

Comment (Giuseppe5) [reply src/brevitas_examples/llm/main.py:498]: https://github.com/EleutherAI/lm-evaluation-harness/blob/6ccd520f3fb2b5d74c6f14c05f9d189521424719/lm_eval/evaluator.py#L93

Looks like it only support torch devices. Even with accelerate, the execution device is always GPU 0, and that's where the data should go. Accelerate will take care of everything else under the hood.

**Pull request merged**